src/Diagnostics.DataProviders/DataProviders/SupportObserverDataProvider.cs

```diff
@@ -226,16 +227,62 @@ public override async Task<dynamic> GetHostingEnvironmentPostBody(string hosting
             return hostingEnvironmentPostBody;
         }
 
+        public override async Task<DataTable> ExecuteSqlQueryAsync(string cloudServiceName, string query)
+        {
+            if (!query.StartsWith("\""))
+            {
+                // BUG: Passing as JSON requires query to be wrapped in quotes, fix API expected content type
+                query = $"\"{query}\"";
+            }
+
+            var request = new HttpRequestMessage(HttpMethod.Post, $"/api/service/{cloudServiceName}/invokesql?api-version=2.0")
+            {
+                Content = new StringContent(query, Encoding.Default, "application/json")
+            };
+
+            var response = await SendObserverRequestAsync(request);
+            var result = await response.Content.ReadAsStringAsync();
+
+            try
+            {
+                response.EnsureSuccessStatusCode();
+            }
+            catch (HttpRequestException)
+            {
+                Logger.LogDataProviderMessage(RequestId, "ObserverDataProvider",
+                    $"message:Observer SQL query request failed, query:{query}, statusCode:{response.StatusCode}, response:{result}");
+                throw;
+            }
+
+            return TryDeserializeDataTable(result);
+        }
+
+        private static DataTable TryDeserializeDataTable(string json)
+        {
+            DataTable datatable;
+            try
+            {
+                datatable = (DataTable)JsonConvert.DeserializeObject(json, typeof(DataTable));
+            }
+            catch (JsonReaderException ex)
+            {
+                throw new Exception($"SQL Query did not return parsable JSON; response: \"{json}\"", ex);
+            }
+
+            return datatable;
+        }
+
         public override HttpClient GetObserverClient()
         {
+            // Instantiating an HttpClient class for every request will exhaust the number of sockets available under heavy loads
```

Comment:
I believe we should fix this as this will cause LSI by leaking sockets.

Reply:
I have a bug task for it. It will probably be a larger refactor throughout the codebase.

Reply:
I think this is the only place we might be creating http client per request. Rest all has a singleton instance or static.

Reply:
True but it is still advised not to directly create an HttpClient. There are a few plugins with Polly that Hawk and I are looking at integrating as well that will automate the exponential backoff and circuit breaker patterns on our requests that will improve the usability of the services as well, so it looks like it will be a pretty involved refactor.

Reply:
Singletons are fine as far as creating an HttpClient, the documentation was saying that in the case of creating our own HttpClient even if it is singleton we need to be weary of DNS changes which honestly we're not to worried about. But I would love to utilize the Polly library for resiliency. It sounds really useful. Good find @dagrooms52